content/kb/tutorials/databases/aws-s3.md

```diff
@@ -55,9 +55,11 @@ access keys. More information [here](https://docs.aws.amazon.com/general/latest/
 
 </Tip>
 
-## Add the key to your local app secrets
+## Set up your AWS credentials locally
 
-Your local Streamlit app will read secrets from a file `.streamlit/secrets.toml` in your app's root directory. Create this file if it doesn't exist yet and add to it the access key ID, access key secret, and the AWS default region you noted down earlier, as shown below:
+Streamlit FilesConnection and s3fs will read and use your existing [AWS credentials and configuration](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html) if available - such as from an `~/.aws/credentials` file or environment variables.
+
+If you don't already have this set up, or plan to host the app on Streamlit Community Cloud, you should specify the credentials from a file `.streamlit/secrets.toml` in your app's root directory or your home directory. Create this file if it doesn't exist yet and add to it the access key ID, access key secret, and the AWS default region you noted down earlier, as shown below:
 
 ```toml
 # .streamlit/secrets.toml
```

Comment:
Shouldn't this be inside a [connections.s3] block?

Reply:
IMO if just adding the AWS keys it's more intuitive to add them at top level as environment variables since I expect the names will be more familiar than the s3fs `key` / `secret` arguments (which I think are sort of odd!). But definitely both are valid. If you think it's better to list it the other way in the tutorial, happy to change it!

Reply:
I just forgot that it works that way, by automatically adding them as environment variables. That sounds good to me, as I think it is probably the most common pattern for AWS creds.